azkaban-common/src/main/java/azkaban/executor/mail/DefaultMailCreator.java

```diff
@@ -130,12 +130,12 @@ public boolean createErrorEmail(ExecutableFlow flow, EmailMessage message,
     if (emailList != null && !emailList.isEmpty()) {
       message.addAllToAddress(emailList);
       message.setMimeType("text/html");
-      message.setSubject("Flow '" + flow.getFlowId() + "' has failed on "
-          + azkabanName);
+      message.setSubject("Flow '" + flow.getFlowId()
+          + "' has failed on " + azkabanName);
 
       message.println("<h2 style=\"color:#FF0000\"> Execution '" + execId
-          + "' of flow '" + flow.getFlowId() + "' has failed on " + azkabanName
-          + "</h2>");
+          + "' of flow '" + flow.getFlowId() + "' of project '"
+          + flow.getProjectName() + "' has failed on " + azkabanName + "</h2>");
```

Comment:
createFirstErrorMessage and createErrorEmail are (from what I can tell) two distinct things. I've never actually seen the first one (createFirstErrorMessage) so for all I know that copy is never seen by users. But we can definitely change it to one/the other and then template it if you think that'd be a good idea!